diff --git a/multidict/_multidict.c b/multidict/_multidict.c
index 73c0022..a7bcd8e 100644
--- a/multidict/_multidict.c
+++ b/multidict/_multidict.c
@@ -275,8 +275,7 @@ multidict_getall(MultiDictObject *self, PyObject *const *args,
             Py_INCREF(_default);
             return _default;
         } else {
-            PyErr_SetObject(PyExc_KeyError, key);
-            return NULL;
+            return PyList_New(0);
         }
     } else {
         return list;
diff --git a/multidict/_multidict_py.py b/multidict/_multidict_py.py
index e286713..54dc44e 100644
--- a/multidict/_multidict_py.py
+++ b/multidict/_multidict_py.py
@@ -663,9 +663,9 @@ class MultiDict(_CSMixin, MutableMultiMapping[_V]):
             for idx in restore:
                 entries[idx].hash = hash_  # type: ignore[union-attr]
             return res
-        if not res and default is not sentinel:
+        if default is not sentinel:
             return default
-        raise KeyError(f"Key not found: {key!r}")
+        return []
 
     @overload
     def getone(self, key: str) -> _V: ...
diff --git a/tests/test_multidict.py b/tests/test_multidict.py
index d62db8c..a0ad66c 100644
--- a/tests/test_multidict.py
+++ b/tests/test_multidict.py
@@ -766,8 +766,7 @@ class TestMultiDict(BaseMultiDictTest):
 
         assert d.getall("key") == ["value1", "value2"]
 
-        with pytest.raises(KeyError, match="some_key"):
-            d.getall("some_key")
+        assert d.getall("some_key") == []
 
         default = object()
         assert d.getall("some_key", default) is default
@@ -856,9 +855,7 @@ class TestCIMultiDict(BaseMultiDictTest):
         assert len(d) == 2
 
         assert d.getall("key") == ["value1", "value2"]
-
-        with pytest.raises(KeyError, match="some_key"):
-            d.getall("some_key")
+        assert d.getall("some_key") == []
 
     def test_get(self, cls: type[CIMultiDict[int]]) -> None:
         d = cls([("A", 1), ("a", 2)])
diff --git a/tests/test_mutable_multidict.py b/tests/test_mutable_multidict.py
index 53b46a8..50954d3 100644
--- a/tests/test_mutable_multidict.py
+++ b/tests/test_mutable_multidict.py
@@ -46,8 +46,7 @@ class TestMutableMultiDict:
 
         assert d.getall("key") == ["value1", "value2"]
 
-        with pytest.raises(KeyError, match="some_key"):
-            d.getall("some_key")
+        assert d.getall("some_key") == []
 
         default = object()
         assert d.getall("some_key", default) is default
@@ -485,9 +484,7 @@ class TestCIMutableMultiDict:
         assert len(d) == 2
 
         assert d.getall("key") == ["value1", "value2"]
-
-        with pytest.raises(KeyError, match="some_key"):
-            d.getall("some_key")
+        assert d.getall("some_key") == []
 
     def test_ctor(
         self,
